src/libs/httpserver/server_http.hpp

```diff
@@ -1,339 +1,526 @@
+        port(port), thread_pool_size(thread_pool_size), timeout_request(timeout_request), timeout_content(timeout_content), address(address) {}
+
+    public:
+      /// Port number to use. Defaults to 80 for HTTP and 443 for HTTPS. Set to 0 get an assigned port.
+      unsigned short port;
+      /// If io_service is not set, number of threads that the server will use when start() is called.
+      /// Defaults to 1 thread.
+      std::size_t thread_pool_size = 1;
+      /// Timeout on request handling. Defaults to 5 seconds.
+      long timeout_request = 5;
+      /// Timeout on content handling. Defaults to 300 seconds.
+      long timeout_content = 300;
+      /// Maximum size of request stream buffer. Defaults to architecture maximum.
+      /// Reaching this limit will result in a message_size error code.
+      std::size_t max_request_streambuf_size = std::numeric_limits<std::size_t>::max();
+      /// IPv4 address in dotted decimal form or IPv6 address in hexadecimal notation.
+      /// If empty, the address will be any address.
+      std::string address;
+      /// Set to false to avoid binding the socket to an address that is already in use. Defaults to true.
+      bool reuse_address = true;
+      /// Make use of RFC 7413 or TCP Fast Open (TFO)
+      bool fast_open = false;
+    };
+    /// Set before calling start().
+    Config config;
 
   private:
-    std::vector<std::pair<
-      std::string,
-      std::vector<std::pair<
-          REGEX_NS::regex,
-          std::function<void(
-              std::shared_ptr<typename ServerBase<socket_type>::Response>,
-              std::shared_ptr<
-                  typename ServerBase<socket_type>::Request>)> > > > >
-      opt_resource;
-
- public:
-  virtual void start() {
-    //Copy the resources to opt_resource for more efficient request processing
-    opt_resource.clear();
-    for(auto& res: resource) {
-        for(auto& res_method: res.second) {
-            auto it=opt_resource.end();
-            for(auto opt_it=opt_resource.begin();opt_it!=opt_resource.end();opt_it++) {
-                if(res_method.first==opt_it->first) {
-                    it=opt_it;
-                    break;
-                }
-            }
-            if(it==opt_resource.end()) {
-                opt_resource.emplace_back();
-                it=opt_resource.begin()+(opt_resource.size()-1);
-                it->first=res_method.first;
-            }
-            it->second.emplace_back(REGEX_NS::regex(res.first), res_method.second);
-        }
+    class regex_orderable : public regex::regex {
+    public:
+      std::string str;
+
+      regex_orderable(const char *regex_cstr) : regex::regex(regex_cstr), str(regex_cstr) {}
+      regex_orderable(std::string regex_str_) : regex::regex(regex_str_), str(std::move(regex_str_)) {}
+      bool operator<(const regex_orderable &rhs) const noexcept {
+        return str < rhs.str;
+      }
+    };
+
+  public:
+    // [CLIPPER]
```

Comment:
I found in the previous code, these were formatted (I believe we ran the formatter before?). We don't use the C++ formatter anymore? @simon-mo @withsmilo